addons/website_event_track/models/event_track.py

```diff
@@ -165,6 +165,14 @@ def _compute_website_url(self):
 
     # STAGES
 
+    @api.depends('stage_id')
+    def _compute_is_visible(self):
```

Comment:
I think this field is not necessary, as it basically depends on ACLs. I mean: you use is_visible_in_agenda, and website published is managed by ACLs (non event users cannot see them). When dealing with sudo in frontend, better keep explicit conditions like you already did (explicitely filter out unpublished). As it is used only once, I suppose this field could be removed and its use made explicit in controllers.